model-deployment/containers/Triton/Deploying onnx and pytorch multi models using Triton.md

```diff
@@ -0,0 +1,364 @@
+{
+name: "input",
+data_type: TYPE_FP32,
+dims: [ 3, 224, 224 ]
+}
+]
+
+output [
+{
+name: "output",
+data_type: TYPE_FP32,
+dims: [ 1000 ]
+}
+]
+
+instance_group [
+{
+kind: KIND_CPU
+}
+]
+```
+
+```
+cp config.pbtxt models/resnet
+```
+
+###Step 1.2  Upload NVIDIA based triton server image to OCI Container Registry (OCIR)
+
+####Creating image locally
+```
+mkdir -p tritonServer
+cd tritonServer
+git clone https://github.com/triton-inference-server/server.git -b v2.30.0 --depth 1
+cd server
+python compose.py --backend onnxruntime --backend pytorch --repoagent checksum --output-name $(OCIR_REGION).ocir.io/$(OCIR_NAMESPACE)/oci-datascience-triton-server/onnx-pytorch-runtime:1.0.0
+```
+Refer  https://docs.oracle.com/en-us/iaas/data-science/using/mod-dep-byoc.htm#construct-container for details on uploading image to OCIR
+
+
+
+###Step 1.3 Upload model artifact to Model catalog
+
+Compress model_repository folder created in Step 1.1 in zip format and upload it to model catalog via python sdk. Refer https://docs.oracle.com/en-us/iaas/data-science/using/models_saving_catalog.htm for details
+
+
+
+###Step 1.4 Create Model Deployment
+OCI Data Science Model Deployment supports Triton Inference Server as a special container, mapping the service-mandated endpoints to the Triton's inference and health HTTP/REST endpoint to free you from having to do so. To Enable it, set the following environment variable when creating the Model Deployment:
+
+```
+CONTAINER_TYPE = TRITON
+```
+
+
+####Using python sdk
+
+```
+# create a model configuration details object
+model_config_details = ModelConfigurationDetails(
+model_id= <model_id>,
+bandwidth_mbps = <bandwidth_mbps>,
+instance_configuration = <instance_configuration>,
+scaling_policy = <scaling_policy>
+)
+
+# create the container environment configuration
+environment_config_details = OcirModelDeploymentEnvironmentConfigurationDetails(
+environment_configuration_type="OCIR_CONTAINER",
+environment_variables={'CONTAINER_TYPE': 'TRITON'},
+image="iad.ocir.io/testtenancy/oci-datascience-triton-server/onnx-runtime:1.0.0",
+image_digest="sha256:aa32690a166b09015d34c9372812ee9c878cbdc75649f7be6e4465b5eb9ad290",
+cmd=[
+"/opt/nvidia/nvidia_entrypoint.sh",
+"tritonserver",
+"--model-repository=/opt/ds/model/deployed_model"
+],
+server_port=8000,
+health_check_port=8000
+)
+
+# create a model type deployment
+single_model_deployment_config_details = data_science.models.SingleModelDeploymentConfigurationDetails(
+deployment_type="SINGLE_MODEL",
+model_configuration_details=model_config_details,
+environment_configuration_details=environment_config_details
+)
+
+# set up parameters required to create a new model deployment.
+create_model_deployment_details = CreateModelDeploymentDetails(
+display_name= <deployment_name>,
+model_deployment_configuration_details = single_model_deployment_config_details,
+compartment_id = <compartment_id>,
+project_id = <project_id>
+)
+```
+
+###Step 1.5 Update Model Deployment
```

Comment:
Update model deployment is not a necessary step. perhaps better move it to a different section in the document (not as a step)?